src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopTerasortExecutor.cs

```diff
@@ -0,0 +1,381 @@
+            this.systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManagement.StateManager;
+            this.packageManager = this.systemManagement.PackageManager;
+            this.fileSystem = this.systemManagement.FileSystem;
+        }
+
+        /// <summary>
+        /// A policy that defines how the component will retry when
+        /// it experiences transient issues.
+        /// </summary>
+        public IAsyncPolicy RetryPolicy { get; set; }
+
+        /// <summary>
+        /// Java Development Kit package name.
+        /// </summary>
+        public string JdkPackageName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(HadoopTerasortExecutor.JdkPackageName));
+            }
+        }
+
+        /// <summary>
+        /// The Row Number count defined in the profile.
+        /// </summary>
+        public int RowCount
+        {
+            get
+            {
+                return this.Parameters.GetValue<int>(nameof(this.RowCount), 100000);
+            }
+
+            set
+            {
+                this.Parameters[nameof(this.RowCount)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The path to the Hadoop package.
+        /// </summary>
+        private string PackageDirectory { get; set; }
+
+        /// <summary>
+        /// Path to the Java Package Directory.
+        /// </summary>
+        private string JavaPackageDirectory { get; set; }
+
+        /// <summary>
+        /// The path to the Hadoop executable file.
+        /// </summary>
+        private string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// Path to the java executable file. 
+        /// </summary>
+        private string JavaExecutablePath { get; set; }
+
+        /// <summary>
+        /// Initializes the environment for execution of the Hadoop Terasort workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.state = await this.stateManager.GetStateAsync<HadoopExecutorState>($"{nameof(HadoopExecutorState)}", cancellationToken)
+                ?? new HadoopExecutorState();
+
+            DependencyPath workloadPackage = await this.packageManager.GetPlatformSpecificPackageAsync(
+                this.PackageName, this.Platform, this.CpuArchitecture, cancellationToken);
+
+            DependencyPath javaPackage = await this.GetPackageAsync(
+                this.JdkPackageName, cancellationToken);
+
+            this.PackageDirectory = workloadPackage.Path;
+            this.JavaPackageDirectory = javaPackage.Path;
+
+            this.JavaExecutablePath = this.PlatformSpecifics.Combine(this.JavaPackageDirectory, "bin", "java");
+
+            switch (this.Platform)
```

Comment:
Here you don't have to do this check, you could do the check in the IsValidate, here you could assume user are on linux